### PR TITLE
Json whitespace

### PR DESCRIPTION
**Pull request by jugglinmike**
@mathiasbynens I planned to re-factor (and rename) the test for `JSON.parse` as
a follow-up to gh-709. As I did so, I came to believe that it wasn't necessary
to remove U+180E from that test in the first place. The second commit
re-introduces that assertion and includes an explanation--can you let me know
what you think?

### CODE REVIEW

File: test/built-ins/JSON/parse/invalid-whitespace.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2012 Ecma International.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 15.12.1.1-0-7
```
Comment (mathiasbynens) [inline test/built-ins/JSON/parse/invalid-whitespace.js]: `esid: sec-json.parse`

**Comment by mathiasbynens** [comment]
You’re right! Sorry for incorrectly modifying that test.

From ECMA-404:

> [In JSON,] Insignificant whitespace is allowed before or after any token. The whitespace characters are: character tabulation (U+0009), line feed (U+000A), carriage return (U+000D), and space (U+0020)

**Comment by jugglinmike** [comment]
No problem at all. Thanks for the review!

**Comment by tcare** [comment]
Looks good.

**Pull request merged**